frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -181,6 +221,66 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
+                        if (!values.featureFlags[FEATURE_FLAGS.SESSION_RECORDING_BLOB_REPLAY]) {
+                            console.error(
+                                'FOR SIMPLICITY THIS MECHANISM ASSUMES YOU HAVE THE BLOB STORAGE FLAG ENABLED'
+                            )
+                            return false
+                        }
+                        if (!values.sessionPlayerSnapshotData?.snapshots.length) {
+                            console.log('There are no current values **from blob storage**')
+                        }
+                        const fakePoint = (): Promise<void> => Promise.resolve()
+                        let comparisonData = await makeSnapshotsAPICall({
+                            breakpoint: fakePoint,
+                            nextUrl: undefined,
+                            recordingStartTime: props.recordingStartTime,
+                            blobLoadingEnabled: false,
+                            currentSnapshotData: null,
+                            currentTeamId: values.currentTeamId,
+                            sessionRecordingId: props.sessionRecordingId,
+                        })
+
+                        while (comparisonData.next) {
+                            comparisonData = await makeSnapshotsAPICall({
+                                breakpoint: fakePoint,
+                                nextUrl: comparisonData.next,
+                                recordingStartTime: props.recordingStartTime,
+                                blobLoadingEnabled: false,
+                                currentSnapshotData: comparisonData,
+                                currentTeamId: values.currentTeamId,
+                                sessionRecordingId: props.sessionRecordingId,
+                            })
+                        }
+                        console.log('finished loading comparison data')
+                        console.log('now we have: ', {
+                            fromBlobStorage: values.sessionPlayerSnapshotData?.snapshots,
+                            fromAPI: comparisonData?.snapshots,
+                        })
+                        const areExactlyTheSame =
+                            (values.sessionPlayerSnapshotData?.snapshots?.length || -1) >= 1 &&
+                            values.sessionPlayerSnapshotData?.snapshots?.every((snapshot, index) => {
+                                const comparisonSnapshot = comparisonData?.snapshots?.[index]
+                                if (!comparisonSnapshot) {
+                                    return false
+                                }
+
+                                // object from blobStorage has a `delay` key that isn't on the API response so...
+                                const { delay, ...storageSnapshot } = snapshot
+
+                                return objectsEqual(storageSnapshot as RecordingSnapshot, comparisonSnapshot)
```

Comment:
Probably not the biggest deal but this will likely hang badly on big recordings. Maybe we just need to compare the timestamps or something?